inflight.go

```diff
@@ -33,40 +33,34 @@ func makeCall[T, V any](ctx context.Context, c *Client[T], key string, fn FetchF
 	}()
 
 	response, err := fn(ctx)
+	call.val = response
```

Comment:
This change is great and I agree with it.

I followed the valuable [discussion](https://github.com/viccon/sturdyc/pull/36#issuecomment-2740511229) that led to do this

But I think the changes introduced by https://github.com/viccon/sturdyc/commit/faee623e33050bb4e37288822939bc981f81fe71 should be convered by tests

Adding such a change is great but it requires to make sure any future changes won't break this behavior

See also https://github.com/viccon/sturdyc/pull/36#discussion_r2007078109

Reply:
Hi, yeah I agree that we should get test coverage for this!